### PR TITLE
add 'use strict' to mod scripts

### PR DESCRIPTION
**Pull request by DeltaNedas**
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode

tl;dr;
`const x = 0; x = 1; print(x);`
will throw an error instead of silently ignoring it and printing 0

### CODE REVIEW

**Comment by zethnest** [comment]
Maybe a warning instead of error?

File: core/src/mindustry/mod/Scripts.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ private boolean run(String script, String file, boolean wrap){
                 context.evaluateString(scope, "modName = \"" + currentMod.name + "\"\nscriptName = \"" + file + "\"", "initscript.js", 1, null);
             }
             context.evaluateString(scope,
-            wrap ? "(function(){\n" + script + "\n})();" : script,
+            wrap ? "(function(){'use strict'\n" + script + "\n})();" : script,
```
Comment (Shawak) [inline core/src/mindustry/mod/Scripts.java:96]: It really should be `'use strict';\n`

Comment (ILiekMelons) [reply core/src/mindustry/mod/Scripts.java:96]: you missed a semicolon

**Comment by Anuken** [comment]
image

This is already a thing...?

**Comment by DeltaNedas** [comment]
That specifically is fixed but there are lots of other things specified in the link

**Pull request merged**